assets/app/scripts/app.js

```diff
@@ -17,24 +18,121 @@ angular
+    };
+
+    var templatePath = "views";
+    var pluginName = "openshiftConsole";
+    var tab = builder.create()
+     .id(builder.join(pluginName, "overview"))
+     .title(function () { return "Overview"; })
+     .template(template)
+     .isSelected(isSelected)
+     .href(projectHref("overview"))
+      .page(function () { return builder.join(templatePath, 'project.html'); })                     
+     .build();
+    tab.icon = "dashboard";
+    OPENSHIFT_BUILT_TABS.push(tab);
+
+
+    tab = builder.create()
+      .id(builder.join(pluginName, "browse"))
+      .title(function () { return "Browse"; })
+      .template(template)
+      .isSelected(isSelected)      
+      .href(projectHref("browse"))
+      .subPath("Builds", "builds", builder.join(templatePath, 'builds.html'))
+      .subPath("Deployments", "deployments", builder.join(templatePath, 'deployments.html'))
+      .subPath("Images", "images", builder.join(templatePath, 'images.html'))
+      .subPath("Pods", "pods", builder.join(templatePath, 'pods.html'))
+      .subPath("Services", "services", builder.join(templatePath, 'services.html'))
+      .build();
+    tab.icon = "sitemap";
+    OPENSHIFT_BUILT_TABS.push(tab);
+
+  }])
+  .config(function ($routeProvider) {
     $routeProvider
       .when('/', {
         templateUrl: 'views/projects.html',
         controller: 'ProjectsController'
       })
       .when('/project/:project', {
-        templateUrl: 'views/project.html',
-        controller: 'ProjectController'
+        redirectTo: function(params) {
+          return '/project/' + params.project + "/overview";
+        }
       })      
-      .when('/pods', {
-        templateUrl: 'views/pods.html',
-        controller: 'PodsController'
+      .when('/project/:project/overview', {
+        templateUrl: 'views/project.html'
       })
+      .when('/project/:project/browse', {
+        redirectTo: function(params) {
+          return '/project/' + params.project + "/browse/pods";  // TODO decide what subtab to default to here
+        }
+      })      
+      .when('/project/:project/browse/builds', {
+        templateUrl: 'views/builds.html'
+      })      
+      .when('/project/:project/browse/deployments', {
+        templateUrl: 'views/deployments.html'
+      })            
+      .when('/project/:project/browse/images', {
+        templateUrl: 'views/images.html'
+      })      
+      .when('/project/:project/browse/pods', {
+        templateUrl: 'views/pods.html'
+      }) 
+      .when('/project/:project/browse/services', {
+        templateUrl: 'views/services.html'
+      })       
       .otherwise({
         redirectTo: '/'
       });
-    $locationProvider.html5Mode(true);
-  });
+  })
+  .run(["HawtioNav", function (HawtioNav) {
+    for (var i = 0; i < OPENSHIFT_BUILT_TABS.length; i++) {
+      HawtioNav.add(OPENSHIFT_BUILT_TABS[i]);
+    }
+  }]);
 
 hawtioPluginLoader.addModule('openshiftConsole');
```

Comment:
hawtio-core, which is already listed as a dependency, and all of the bower dependency js files are loaded prior to our app